xcp_d/interfaces/connectivity.py

```diff
@@ -119,23 +140,321 @@ def _run_interface(self, runtime):
+
+        timeseries_df.to_csv(self._results["timeseries"], sep="\t", na_rep="n/a", index=False)
+        correlations_df.to_csv(
+            self._results["correlations"],
+            sep="\t",
+            na_rep="n/a",
+            index_label="Node",
+        )
+        coverage_df.to_csv(self._results["coverage"], sep="\t", index_label="Node")
+
+        return runtime
+
+
+class _CiftiConnectInputSpec(BaseInterfaceInputSpec):
+    min_coverage = traits.Float(
+        default=0.5,
+        usedefault=True,
+        desc=(
+            "Coverage threshold to apply to parcels. "
+            "Any parcels with lower coverage than the threshold will be replaced with NaNs. "
+            "Must be a value between zero and one. "
+            "Default is 0.5."
+        ),
+    )
+    data_file = File(
+        exists=True,
+        mandatory=True,
+        desc="Dense CIFTI time series file to parcellate.",
+    )
+    atlas_file = File(
+        exists=True,
+        mandatory=True,
+        desc=(
+            "Atlas CIFTI file to use to parcellate data_file. "
+            "This file must already be resampled to the same structure as data_file."
+        ),
+    )
+    parcellated_atlas = File(
+        exists=True,
+        mandatory=True,
+        desc=(
+            "Atlas CIFTI that has been parcellated with itself to make a .pscalar.nii file. "
+            "This is just used for its ParcelsAxis."
+        ),
+    )
+    atlas_labels = File(exists=True, mandatory=True, desc="atlas labels file")
+
+
+class _CiftiConnectOutputSpec(TraitedSpec):
+    coverage_pscalar = File(exists=True, mandatory=True, desc="Coverage CIFTI file.")
+    ptseries = File(exists=True, mandatory=True, desc="Parcellated data CIFTI file.")
+    pconn = File(exists=True, mandatory=True, desc="Correlation matrix pconn.nii file.")
+    coverage = File(exists=True, mandatory=True, desc="Coverage tsv file.")
+    timeseries = File(exists=True, mandatory=True, desc="Parcellated data tsv file.")
+    correlations = File(exists=True, mandatory=True, desc="Correlation matrix tsv file.")
+
+
+class CiftiConnect(SimpleInterface):
+    """Extract timeseries and compute connectivity matrices.
+
+    Write out time series using Nilearn's NiftiLabelMasker
+    Then write out functional correlation matrix of
+    timeseries using numpy.
+    """
+
+    input_spec = _CiftiConnectInputSpec
+    output_spec = _CiftiConnectOutputSpec
 
-        # Use numpy for correlation matrix
-        correlation_matrices = np.corrcoef(time_series.T)
+    def _run_interface(self, runtime):
+        coverage_threshold = self.inputs.min_coverage
+        data_file = self.inputs.data_file
+        atlas_file = self.inputs.atlas_file
+        pscalar_file = self.inputs.parcellated_atlas
+        atlas_labels = self.inputs.atlas_labels
+
+        cifti_intents = get_cifti_intents()
+
+        assert data_file.endswith(".dtseries.nii"), data_file
```

Comment:
won't `mandatory=True` enforce that data_file is defined?

Reply:
True, but I wanted to make sure that only dtseries.nii files were fed into this interface.